ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java

```diff
@@ -0,0 +1,104 @@
+package com.parse;
+
+import android.util.Log;
+
+import java.net.URI;
+import java.util.Locale;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+
+/* package */ class OkHttp3WebSocketClient implements WebSocketClient {
+
+    private static final String LOG_TAG = "OkHttpWebSocketClient";
+
+    private final WebSocketClientCallback webSocketClientCallback;
+    private okhttp3.WebSocket webSocket;
+    private State state = State.NONE;
+    private OkHttpClient client;
+    private final String url;
+    private final int STATUS_CODE = 200;
+    private final String CLOSING_MSG = "User invoked close";
+
+    private final WebSocketListener handler = new WebSocketListener() {
+        @Override
+        public void onOpen(okhttp3.WebSocket webSocket, Response response) {
+            setState(State.CONNECTED);
+            webSocketClientCallback.onOpen();
+        }
+
+        @Override
+        public void onMessage(okhttp3.WebSocket webSocket, String text) {
+            webSocketClientCallback.onMessage(text);
+        }
+
+        @Override
+        public void onMessage(okhttp3.WebSocket webSocket, ByteString bytes) {
+            Log.w(LOG_TAG, String.format(Locale.US, "Socket got into inconsistent state and received %s instead.", bytes.toString()));
+        }
+
+        @Override
+        public void onClosed(okhttp3.WebSocket webSocket, int code, String reason) {
+            setState(State.DISCONNECTED);
+            webSocketClientCallback.onClose();
+        }
+
+        @Override
+        public void onFailure(okhttp3.WebSocket webSocket, Throwable t, Response response) {
+            webSocketClientCallback.onError(t);
+        }
+    };
+
+    private OkHttp3WebSocketClient(WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+        this.webSocketClientCallback = webSocketClientCallback;
+        client = new OkHttpClient();
```

Comment:
Mmm not sure to understand. Maybe my comment wasn't clear.
Why did you add the ability to pass a OkHttpClient? Is it useful?